src/blocks/Block.as

```diff
@@ -727,14 +669,18 @@ public class Block extends Sprite {
 	}
 
 	public function topBlock():Block {
-		var result:DisplayObject = this;
-		while (result.parent is Block) result = result.parent;
-		return Block(result);
+		var result:Block = this;
+		while (result.parent is Block)
+			result = result.parent as Block;
```

Comment:
I believe this is logically the same but I switched DisplayObject to Block which makes it stricter and matches the return value. This code will crawl up nested arguments if called on a reporter Block. The final logic will utilize the BlockStack if present to get the top block reference.